pkgs/applications/misc/mepo/default.nix

```diff
@@ -16,26 +16,53 @@
 , xdotool
 , bemenu
 , withX11 ? false
+
+# transmits user's location without asking first
+, geoclueSupport ? false , geoclue2
+
+# transmits user's location without asking first
+, reportLocationToMozilla ? false
+
+, gpsdSupport ? true , gpsd
 }:
 
 let
   menuInputs = if withX11 then [ dmenu xdotool ] else [ bemenu ];
 in stdenv.mkDerivation rec {
   pname = "mepo";
-  version = "0.4.2";
+  version = "0.5";
 
   src = fetchFromSourcehut {
     owner = "~mil";
     repo = pname;
     rev = version;
-    hash = "sha256-k6YXaqB3EwbDPlTvijZf10q+IYwt4/MiqGXL495KIcY=";
+    hash = "sha256-V8NdlDnj6nyC7pAPu76tMbw//5LqgPcd0jOKSsvx9ZU=";
   };
 
   nativeBuildInputs = [ pkg-config zig makeWrapper ];
 
   buildInputs = [
     curl SDL2 SDL2_gfx SDL2_image SDL2_ttf inconsolata-nerdfont jq ncurses
-  ] ++ menuInputs;
+  ] ++ menuInputs
+    ++ lib.optional geoclueSupport geoclue2
+    ++ lib.optional gpsdSupport gpsd;
+
+  patches = lib.optionals (!reportLocationToMozilla) [
+    ./dont-report-user-location-to-mozilla-without-asking.patch
+  ] ++ lib.optionals (!geoclueSupport) [
+    ./dont-report-user-location-to-geoclue-without-asking.patch
+  ];
+
+  postPatch = lib.optionalString gpsdSupport ''
+    substituteInPlace \
+      scripts/mepo_ui_menu_user_pin_updater.sh \
+      --replace gpspipe ${gpsd}/bin/gpspipe
+  '' + lib.optionalString geoclueSupport ''
+    substituteInPlace \
+      scripts/mepo_ui_menu_user_pin_updater.sh \
+      --replace /usr/libexec/geoclue-2.0/demos/ \
+                ${geoclue2}/libexec/geoclue-2.0/demos/
```

Comment:
```suggestion
    substituteInPlace scripts/mepo_ui_menu_user_pin_updater.sh \
      --replace /usr/libexec/geoclue-2.0/demos/ ${geoclue2}/libexec/geoclue-2.0/demos/
```

Reply:
I think this is a case where the cost of sticking to 80 columns is low enough to be worthwhile.